io/driver/module_unload_load.py

```diff
@@ -0,0 +1,139 @@
+        """
+        Unloading and loading the given module
+        """
+        for mod1 in module_list:
+            if linux_modules.module_is_loaded(mod1) is False:
+                linux_modules.load_module(mod1)
+                time.sleep(self.load_unload_sleep_time)
+
+        for _ in range(0, self.iteration):
+            for mdl in module_list:
+                sub_mod = self.get_depend_modules(mdl)
+                if sub_mod:
+                    sub_mod = sub_mod.split(' ')
+                    for mod in sub_mod:
+                        if mod == 'multipath':
+                            process.system("multipath -F", ignore_status=True)
+                            cmd = "lsmod | grep -i ^%s" % mdl
+                            for _ in range(20):
+                                time.sleep(3)
+                                out = process.getoutput(cmd).split(" ")[-1]
+                                if out.strip() != '0':
+                                    self.log.info("command value : %s\n" % out)
+                                    process.system("multipath -F",
+                                                   ignore_status=True)
+                                    continue
+                                else:
+                                    break
+                        else:
+                            self.log.info("unloading sub module %s " % mod)
+                            process.system("rmmod %s" % mod)
+                self.log.info("unloading module %s " % mdl)
+                process.system("rmmod %s" % mdl)
+                time.sleep(self.load_unload_sleep_time)
+                if linux_modules.module_is_loaded(mdl) is True:
+                    self.error_modules.append(mdl)
+                self.log.info("loading module : %s " % mdl)
+                linux_modules.load_module(mdl)
+                time.sleep(self.load_unload_sleep_time)
+                if linux_modules.module_is_loaded(mdl) is False:
+                    self.error_modules.append(mdl)
+
+    def test(self):
+        """
+        Begining the test here
+        """
+        pci_addrs = []
+        if self.module:
+            for modl in self.module:
+                self.mod_list.append(modl)
+        elif self.only_io is True:
+            pci_addrs = pci.get_pci_addresses()
+            for pci1 in pci_addrs:
+                self.mod_list.append(pci.get_driver(pci1))
+        else:
+            cmd = "find /lib/modules/%s/ -name /*.ko" % self.uname
+            for line in process.getoutput(cmd).splitlines():
+                driver = process.getoutput(line.split('/')[-1])
+                self.mod_list.append(driver.split('.')[0])
+
+        for mod in self.mod_list:
+            if self.built_in_module(mod) is True:
+                self.mod_list.remove(mod)
+        self.module_load_unload(self.mod_list)
+
+        if self.error_modules:
+            self.error("few modules failed to load unload, please check logs")
```

Comment:
self.fail or self.warn is better here

Reply:
sure will change this to self.fail or self.warn